include/RED4ext/Scripting/Natives/Vector2.hpp

```diff
@@ -0,0 +1,109 @@
+#pragma once
+
+#include <RED4ext/Common.hpp>
+#include <cstdint>
+
+namespace RED4ext
+{
+struct Vector2
+{
+    static constexpr const char* NAME = "Vector2";
+    static constexpr const char* ALIAS = NAME;
+
+    Vector2()
+        : X(0)
+        , Y(0)
+    {
+    }
+
+    Vector2(float x, float y)
+        : X(x)
+        , Y(y)
+    {
+    }
+
+    inline Vector2& operator=(const Vector2& aOther)
+    {
+        X = aOther.X;
+        Y = aOther.Y;
+
+        return *this;
+    }
+
+    inline Vector2 operator+(const Vector2& aOther) const
+    {
+        return {X + aOther.X, Y + aOther.Y};
+    }
+
+    inline Vector2 operator-() const
+    {
+        return {-X, -Y};
+    }
+
+    inline Vector2 operator-(const Vector2& aOther) const
+    {
+        return {X - aOther.X, Y - aOther.Y};
+    }
+
+    inline Vector2 operator*(const float aScale) const
```

Comment:
I think `aScalar` would be better here.

```suggestion
    inline Vector2 operator*(const float aScalar) const
```